src/Provider/AlgoliaPlaces/AlgoliaPlaces.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\AlgoliaPlaces;
+
+use Geocoder\Collection;
+use Geocoder\Exception\InvalidArgument;
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Exception\InvalidServerResponse;
+use Geocoder\Exception\QuotaExceeded;
+use Geocoder\Exception\UnsupportedOperation;
+use Geocoder\Http\Provider\AbstractHttpProvider;
+use Geocoder\Model\Address;
+use Geocoder\Model\AddressBuilder;
+use Geocoder\Model\AddressCollection;
+use Geocoder\Provider\Provider;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Http\Client\HttpClient;
+
+class AlgoliaPlaces extends AbstractHttpProvider implements Provider
+{
+    const TYPE_CITY = 'city';
+
+    const TYPE_COUNTRY = 'country';
+
+    const TYPE_ADDRESS = 'address';
+
+    const TYPE_BUS_STOP = 'busStop';
+
+    const TYPE_TRAIN_STATION = 'trainStation';
+
+    const TYPE_TOWN_HALL = 'townhall';
+
+    const TYPE_AIRPORT = 'airport';
+
+    /** @var string */
+    const ENDPOINT_URL = 'http://places-dsn.algolia.net/1/places/query';
```

Comment:
Do we really want this? If you do not mind I suggest to remove it.

Reply:
in fact in a lot of provider there is a constant like this one

GMaps
https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/GoogleMaps/GoogleMaps.php#L37

For OSM it's a little different but near the same way
https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/GoogleMaps/GoogleMaps.php#L37

TomTom
https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/GoogleMaps/GoogleMaps.php#L37

So I think we can keep it

Reply:
I mean we should not use HTTP but only HTTPS. Im fine with you using a constant.

Reply:
oh, ok. I was not understood your question. So yes I'm totally agree with I will remove it